whistl/util.py

```diff
@@ -5,6 +5,60 @@
 import random
 
 
+def add_genes_to_results(results, gene_file):
+    '''Add the ids of the genes used to train the model to the results dictionary
+
+    Arguments
+    ---------
+    results: dict
+        The dictionary containing metrics about the run
+    gene_file: str or Path
+        The file containing the list of genes to train the model on
+
+    Returns
+    -------
+    results: dict
+        The dictionary passed in with the list of genes added
+    '''
+    genes = parse_gene_file(gene_file)
+    results['genes'] = genes
+
+    return results
+
+
+def add_study_ids_to_results(results, train_dirs, tune_dirs):
+    '''Add the ids of the training and tuning set studies used to the results dictionary
+
+    Arguments
+    ---------
+    results: dict
+        The dictionary containing metrics about the run
+    train_dirs: list of strings
+        The paths to each directory containing a study that was used in training the model
+    tune_dirs: list of strings
+        The paths to each directory containing a study that was used in tuning the model
+
+    Returns
+    -------
+    results: dict
+        The dictionary passed in, with the train and tune study ids added
+    '''
+    train_ids = []
+    tune_ids = []
+
+    for dir_ in train_dirs:
+        study_id = dir_.strip('/').split('/')[-1]
```

Comment:
Since this is a filepath I'd recommend using [os.path.split](https://docs.python.org/3/library/os.path.html#os.path.split) here. I know it's unlikely you'll ever run this on another OS, but it's a good habit to get into for cross-platform compatibility in general.

If you prefer an OO interface, you could try using [pathlib](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parts) for this instead. `pathlib` and `os` have most of the same functionality, just a matter of which interface you prefer.

Reply:
Good call. One note, os.path.split assumes trailing slashes are intentional, so if you're looking at directories you have to use os.path.normpath first.